src/components/Api.js

```diff
@@ -0,0 +1,115 @@
+import { routeData as routes } from "../utils/constants";
+const debug = false;
+// if(debug) console.log(``);
+
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._authorization = headers.authorization;
+    this._contentType = headers["Content-Type"];
+  }
+
+  _buildRequest(routeData, body = undefined) {
+    const url = this._baseUrl + routeData.route;
+    const info = {
+      method: routeData.method,
+      headers: {
+        authorization: this._authorization,
+        "Content-Type": this._contentType,
+      },
+    };
+
+    if (body) info.body = JSON.stringify(body);
+    if (debug) console.log(`call info:`);
+    if (debug) console.log(info);
+
+    return { url: url, info: info };
+  }
+
+  request(routeData, body) {
+    const requestInfo = this._buildRequest(routeData, body);
+    return fetch(requestInfo.url, requestInfo.info)
+      .then((res) => {
+        if (res.ok) {
+          return res.json();
+        }
+        return Promise.reject(`API request Error: ${res.status}`);
```

Comment:
```js
.then(res => {
                if (res.ok) {
                    return res.json();
                }
                return Promise.reject(`Error ${res.status}`);
            })

```

The code of `ok` checking is repeated in **every** request of the project. To avoid it you need to make a special method `_checkResponse` that will check any response from the server if it’s ok or not.


```js
_checkResponse(res) {
        // here is the code of the checking
    }

```

And now you can replace redundant repeating with the only line:


```js
 .then(this._checkResponse)

```

Please, pay attention that you need to pass only the **reference** for the method rather than the call. 

**Could be improved**

You can make a special method (function ) for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.

Reply:
Neat. Thank you!